### PR TITLE
feat: adds initial configuration code

### PR DESCRIPTION
**Pull request by yowainwright**
## Description

This pull request adds initial configuration code. 

> *Everything is implemented for discussion!!! 🙂*

### Configurations added

- Chomp!
- Vitest
- Eslint
- Prettier
- ~~Editorconfig~~
- Devcontainers
- ~~Bun~~ will remove to match jspm patterns
- Typescript
- Commitlint
- ~~Setup Config bash (bin/setup.sh)~~ went with devcontainers only
- Husky
- ~~Lint-staged~~ will remove to match jspm patterns, 

### Todos

- [x] add chomp to `devcontainer` + `bin/setup.sh`
- [x] prefer chomp over npm scripts *(seems like that's the pattern)*
- [ ] fully test `loader.ts`

---

## Discussion

After discussion I will implement a function with tests for `extractPackageNameAndVersion`. 👌 

- Happy to break out pieces that are desired. 👍 
- Happy to remove pieces that are not. 👍 

----

<p align=center>Thank you!!! 🎈🎈🎈</p>

### CODE REVIEW

**Comment by yowainwright** [comment]
Updated 
- Feel free to check it out! 
- Happy to change whatever! 
- I will write tests in a bit.

**Comment by JayaKrishnaNamburu** [comment]
Hey @yowainwright nice changes to the repo. Sorry for the delay to catchup on this. Changes look good, but i don't understand the use of `devcontainers` and the `setup.sh` files. And if you are ok, can we drop the `.editorconfig` ?

**Comment by yowainwright** [comment]
@JayaKrishnaNamburu removed `/bin`. Wrote doc for running the `devcontainer`. Let me know your thoughts!

- [ ] Will wrap tests ASAP!

**Comment by JayaKrishnaNamburu** [comment]
Looking good, can we drop this too from github actions https://codesee.io

**Comment by yowainwright** [comment]
@JayaKrishnaNamburu done!

**Comment by JayaKrishnaNamburu** [comment]
Nice, let me know when you are done with the tests and stuff. We can merge this 👍

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+This project is currently slowly iterating towards a v1 release!
+
+| Version | Supported          |
+| ------- | ------------------ |
+| 0.1.x   | :white_check_mark: |
+
+## Reporting a Vulnerability
+
+To report a Vulnerability, submit an [issue](https://github.com/yowainwright/merge-tsconfig/issues) or [pull request](https://github.com/yowainwright/merge-tsconfig/pulls)!
```
Comment (JayaKrishnaNamburu) [inline .github/SECURITY.md:13]: We can change this to the main repository 👍

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,10 @@
+MIT License
+-----------
+
+Copyright (C) 2023 JayaKrishnaNamburu
```
Comment (JayaKrishnaNamburu) [inline LICENSE:4]: @guybedford should we change this to the @jspm org ?

File: chompfile.toml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+version = 0.1
+
+[[task]]
+name = 'build'
+run = 'tsup src/loader.ts --format esm,cjs --dts'
```
Comment (JayaKrishnaNamburu) [inline chompfile.toml:5]: We can add `deps` and add `npm:install`. So, if we run `chomp` by default. The build will make sure the deps are also installed 👍

**Review by JayaKrishnaNamburu — APPROVED** [review]
Thanks for all the awesome work @yowainwright 🎉 
Really appreciate it 😄

**Pull request merged**